src/components/DevModePanel/utils.ts

```diff
@@ -0,0 +1,21 @@
+import { createGraphiQLFetcher, FetcherOpts } from "@graphiql/toolkit";
+import { createFetch } from "@saleor/sdk";
+
+const authHeaders = ["Authorization", "Authorization-Bearer"];
+
+const authorizedFetch = createFetch();
+
+export const getFetcher = (opts: FetcherOpts) => {
+  let httpFetch = authorizedFetch;
+
+  const hasAuthorizationHeaders = opts.headers && authHeaders.some(header => opts.headers![header]);
+
+  if (hasAuthorizationHeaders) {
+    httpFetch = fetch;
```

Comment:
suggestion: Remember that `createFetch ` from SDK contains features like autoTokenRefresh and refreshOnUnauthorized. IMO would be good to implement at least refreshing the token when expires.

Reply:
If we're talking about app tokens here (the original ticket stems form user trying to use `appToken`) then they don't expire so there is nothing to refresh. https://docs.saleor.io/developer/extending/apps/architecture/server-and-client-side-calls#tokens-comparison